lib/money/money.rb

```diff
@@ -230,11 +240,19 @@ def to_s(style = nil)
   end
 
   def to_json(options = {})
-    to_s
+    if Money.config.legacy_json_format
+      to_s
+    else
+      as_json.to_json
+    end
   end
 
-  def as_json(*args)
-    to_s
+  def as_json(*_args)
```

Comment:
What do you think about [allowing this to be set per call site](https://github.com/Shopify/money/pull/231#issuecomment-802468335), falling back to the global default if that's not done? This allows for more granular transition.

Reply:
`as_json` already accepts [many options](https://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json) `root`, `include`, `except`, etc. We'd need to make our option as part of the hash in order to not break the signature. If we're comfortable with potential key conflicts, then we can go ahead